### PR TITLE
Add CRUD Voice API resource for Call

### PR DESCRIPTION
**Pull request by cubny**
Implements basic CRUD functionality for `calls` endpoint based on [API docs](https://developers.messagebird.com/api/voice-calling#calls)

also note that due to inconsistency of `Call`'s response with the rest of resources I had to implement a specific response handler on top of the existing one.

### CODE REVIEW

File: lib/messagebird/call_list.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'messagebird/list'
+
+class CallList < List
+
+  def data=(value)
+    @items = value.map { |i| @type.new i }
```
Comment (epels) [inline lib/messagebird/call_list.rb]: What do you think about propagating this to `List`'s `items=` method and documenting why things are this way?

File: spec/call_spec.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      "createdAt": "2017-02-16T10:52:00Z",
+      "updatedAt": "2017-02-16T10:59:04Z",
+      "endedAt": "2017-02-16T10:59:04Z"
+    }
+  ],
+  "_links": {
+    "self": "/calls/f1aa71c0-8f2a-4fe8-b5ef-9a330454ef58"
+  }
+}')
+    client.call_view(call_id)
```
Comment (epels) [inline spec/call_spec.rb]: Can you please check the returned call and its attributes? Right now, we're only testing that the HTTP client receives the request we expect it to, but not whether the MessageBird::Client properly deserialises the response.

File: lib/messagebird/client.rb [hunk 3]
```diff
@@ -235,6 +237,25 @@ def voice_message_create(recipients, body, params={})
...
     end
 
+    def call_create(source, destination, call_flow, params={})
+      Call.new(request(
+        :post,
+        'calls',
+        params.merge({ :source => source, :destination => destination, :callFlow => call_flow.to_s })))
+    end
+
+    def call_list
```
Comment (epels) [inline lib/messagebird/client.rb]: The docs aren't entirely clear on this but judging from the [examples](https://developers.messagebird.com/api/voice-calling#calls), it appears this endpoint supports pagination. Maybe it's good to figure out how they do it and then support it here.

File: lib/messagebird/client.rb [hunk 4]
```diff
@@ -235,6 +237,25 @@ def voice_message_create(recipients, body, params={})
         params.merge({ :recipients => recipients, :body => body.to_s })))
     end
 
+    def call_create(source, destination, call_flow, params={})
```
Comment (epels) [inline lib/messagebird/client.rb]: It seems this expects `call_flow` to be a JSON payload already, but I think it makes sense to make it an object: the idea of the SDK is to abstract implementation details like the underlaying transport away.

File: spec/call_spec.rb (hunk 2 above)
Comment (cubny) [reply spec/call_spec.rb]: done!

File: lib/messagebird/client.rb (hunk 3 above)
Comment (cubny) [reply lib/messagebird/client.rb]: done!

File: lib/messagebird/call_list.rb (hunk 1 above)
Comment (cubny) [reply lib/messagebird/call_list.rb]: done!

File: lib/messagebird/client.rb (hunk 4 above)
Comment (cubny) [reply lib/messagebird/client.rb]: good idea! implemented.

File: lib/messagebird/call.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+require 'messagebird/base'
+require 'messagebird/call/webhook'
+
+module MessageBird
+  class Call < MessageBird::Base
+    attr_accessor :id, :status, :source, :destination, :createdAt, :updatedAt, :endedAt, :webhook
```
Comment (khrt) [inline lib/messagebird/call.rb]: It seems the list is missing `callFlow`.

Comment (cubny) [reply lib/messagebird/call.rb]: done

**Pull request merged**